src/account/account.service.ts

```diff
@@ -748,4 +751,51 @@ export class AccountService {
 
         await this.accountRepository.save(updated);
     }
+
+    async getAccountByInboxUrl(inboxUrl: URL): Promise<Account | null> {
+        return await this.accountRepository.getByInboxUrl(inboxUrl);
+    }
+
+    async recordDeliveryFailure(
+        accountId: number,
```

Comment:
If this took an `inboxUrl` we wouldn't have to expose a `getAccountByInboxUrl` method